gnmi/security.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmi
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/tls/certprovider/pemfile"
+	"google.golang.org/grpc/security/advancedtls"
+)
+
+const (
+	credsRefreshDuration = 24 * time.Hour
+)
+
+func gRPCSecurityOption(cfg *Config) ([]grpc.ServerOption, error) {
```

Comment:
Should this be a bit more explicit? If one had a config error here then you could accidentally end up with just TLS vs. mTLS? It seems like the config should probably explicitly say what it is that the config is requesting and then a validation error should be thrown if the incorrect artifacts are provided to achieve that?

Reply:
it's explicit now.